app/services/catalog/discard_restore.rb

```diff
@@ -0,0 +1,47 @@
+module Catalog
+  module DiscardRestore
+    extend ActiveSupport::Concern
+
+    CHILD_DISCARD_TIME_LIMIT = 30
+
+    module ClassMethods
+      def destroy_dependencies(*klass_names)
+        klass_names.each do |klass|
+          define_method "discard_#{klass}" do
```

Comment:
Do we need to include a transaction in this callback? Or is it the `discard` line 13 which provides our integrity?

Reply:
The `(&:discard).any?` `false` check should be handling if something happens. This gets ran before the discard of the object in question actually happens, so if anything happens while discarding children, it will bail.